Godeps/_workspace/src/k8s.io/kubernetes/pkg/securitycontextconstraints/util/util.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+func GetAllFSTypesExcept(exceptions... string) sets.String {
+	fstypes := GetAllFSTypesAsSet()
+	for _, e := range exceptions {
+		fstypes.Delete(e)
+	}
+	return fstypes
+}
+
+func GetAllFSTypesAsSet() sets.String {
+	fstypes := sets.NewString()
+	fstypes.Insert(
+		string(api.FSTypeHostPath),
+		string(api.FSTypeAzureFile),
+		string(api.FSTypeFlocker),
+		string(api.FSTypeFlexVolume),
+		string(api.FSTypeEmptyDir),
+		string(api.FSTypeGCEPersistentDisk),
+		string(api.FSTypeAWSElasticBlockStore),
+		string(api.FSTypeGitRepo),
+		string(api.FSTypeSecret),
+		string(api.FSTypeNFS),
+		string(api.FSTypeISCSI),
+		string(api.FSTypeGlusterfs),
+		string(api.FSTypePersistentVolumeClaim),
+		string(api.FSTypeRBD),
+		string(api.FSTypeCinder),
+		string(api.FSTypeCephFS),
+		string(api.FSTypeDownwardAPI),
+		string(api.FSTypeFC),
+		string(api.FSTypeConfigMap))
+	return fstypes
+}
+
+// getVolumeFSType gets the FSType for a volume.
+func GetVolumeFSType(v api.Volume) (api.FSType, error) {
```

Comment:
does volume validation ensure we'll never have more than one of these set (it should)? don't want to short circuit here and allow a pod through when it specifies two volume types in a single volume